### PR TITLE
Add a new Packages Panel

### PR DESCRIPTION
**Pull request by antograssiot**
This add the packages Panel as discussed  in #478 

For now it is dead simple, just parsing the `composer.lock` but I love the idea proposed by @lorenzo to get an up-to-date report or something. I will probably try to see how it can be done in the next weeks.

Is this ok to target master ?

b500f8b4-dbe2-11e6-83d9-b6063db7f361

### CODE REVIEW

**Comment by antograssiot** [comment]
I don'the like it as is, I will push an update later to remove constants and therefore title from he controller and passed two variable instead of a huge associated array. Nevermore for the loop duplication in the view.

**Comment by lorenzo** [comment]
@antograssiot it is ok to merge in master

File: src/Template/Element/packages_panel.ctp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                <th>Name</th>
+                <th>Version</th>
+            </tr>
+            </thead>
+            <tbody>
+            <?php foreach ($requirements as $package): ?>
+                <?php extract($package); ?>
+                <tr>
+                    <td title="<?= $description ?>">
+                        <a href="https://packagist.org/packages/<?= $name ?>" title="<?= $description ?>" target="_blank" class="package-link"><?= $name ?></a>
```
Comment (markstory) [inline src/Template/Element/packages_panel.ctp]: Should do HTML escaping on `name` and `description`.

File: src/Template/Element/packages_panel.ctp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    </td>
+                    <td>
+                        <span class="package-version"><span class="panel-summary"><?= $version ?></span></span>
```
Comment (markstory) [inline src/Template/Element/packages_panel.ctp]: More HTML escaping here.

File: src/Template/Element/packages_panel.ctp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+/**
+ * @type \DebugKit\View\AjaxView $this
+ * @type array $packages
+ */
+?>
+
+<?php if (!empty($packages)): ?>
```
Comment (markstory) [inline src/Template/Element/packages_panel.ctp]: Should we roll the current version into this panel, and clean up the toolbar state a bit?

Comment (antograssiot) [inline src/Template/Element/packages_panel.ctp]: The current version will be here indeed so if you prefer to remove it from the toolbar it works for me.

Comment (markstory) [inline src/Template/Element/packages_panel.ctp]: I don't think we need both.

Comment (antograssiot) [inline src/Template/Element/packages_panel.ctp]: Ok I'll do that. Just reverting the other PR should be enough

**Comment by antograssiot** [comment]
ok I've done some cleanup and included the requested changes (HTML escaping and remove the recently added CakePHP version from the toolbar)

capture d ecran 2017-01-18 a 06 09 20

**Comment by markstory** [comment]
Great work @antograssiot I'll get this merged and tagged tonight.

**Pull request merged**